### ISSUE TITLE
[FEA] Support element-wise not operator (~) on Bool columns

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#1504]
**Is your feature request related to a problem? Please describe.**
I'd like to be able to call `~series` on Boolean series, like in pandas.

**Describe the solution you'd like**
```python
import cudf
import pandas as pd

s = cudf.Series([True, False, False])
ps = s.to_pandas()

print(ps)
print(~ps)
print(~s)
0     True
1    False
2    False
dtype: bool
0    False
1     True
2     True
dtype: bool
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-292-c58f31f5ac53> in <module>
      7 print(ps)
      8 print(~ps)
----> 9 print(~s)

/conda/envs/cudf/lib/python3.7/site-packages/cudf-0.7.0.dev0+565.gf6ad6de.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/series.py in __invert__(self)
    573         else:
    574             raise TypeError(
--> 575                 f"Operation `~` not supported on {self.dtype.type.__name__}"
    576             )
    577 

TypeError: Operation `~` not supported on bool_
```

**Comment by harrism** [comment]
Should be supported on C++ side if not now than after #1142.

**Comment by beckernick** [comment]
Resolved. Closing.
